engines/scumm/he/basketball/collision/bball_collision.cpp

```diff
@@ -0,0 +1,568 @@
+
+		// Read in this object's description...
+		int descriptionStringLength = objectFile.readUint32LE();
+		char *tmp = (char *)malloc(descriptionStringLength + 1);
+
+		assert(tmp);
+
+		objectFile.read(tmp, descriptionStringLength);
+		tmp[descriptionStringLength] = '\0';
+
+		Common::String tmp2(tmp);
+
+		_vm->_basketball->_court->_objectList[i]._description = tmp2;
+
+		free(tmp);
+
+		// Read in all other object attributes...
+		currentObject->_objectType = (EObjectType)objectFile.readUint32LE();
+		currentObject->_collisionEfficiency = objectFile.readUint32LE();
+		currentObject->_friction  = objectFile.readUint32LE();
+		currentObject->_soundNumber = objectFile.readUint32LE();
+		currentObject->_objectID = objectFile.readUint32LE();
+		currentObject->minPoint.x = objectFile.readUint32LE();
+		currentObject->minPoint.y = objectFile.readUint32LE();
+		currentObject->minPoint.z = objectFile.readUint32LE();
+		currentObject->maxPoint.x = objectFile.readUint32LE();
+		currentObject->maxPoint.y = objectFile.readUint32LE();
+		currentObject->maxPoint.z = objectFile.readUint32LE();
+		objectPtrList[i] = currentObject;
+
+		// Decide if this is a backboard, and keep track of it if it is...
+		if (currentObject->_objectType == kBackboard) {
+			// See which backboard it is...
+			if (((currentObject->minPoint.x + currentObject->maxPoint.x) / 2) < (MAX_WORLD_X / 2)) {
+				_vm->_basketball->_court->_backboardIndex[LEFT_BASKET] = i;
+			} else {
+				_vm->_basketball->_court->_backboardIndex[RIGHT_BASKET] = i;
+			}
+		}
+	}
+
+	_vm->_basketball->_court->_objectTree.initialize(objectPtrList);
+	
+
+	// Lower all the shields...
+	u32_userLowerShields(ALL_SHIELD_ID);
+
+	return 1;
+}
+
+int LogicHEBasketball::u32_userDeinitCourt() {
+	// Make sure the collision object tree has been cleared...
+	_vm->_basketball->_court->_objectTree.~CCollisionObjectTree();
+
+	// Lower all the shields...
+	u32_userLowerShields(ALL_SHIELD_ID);
+
+	return 1;
+}
+
+int LogicHEBasketball::u32_userInitBall(U32FltPoint3D &ballLocation, U32FltVector3D &bellVelocity, int radius, int ballID) {
+	_vm->_basketball->_court->_basketBall._description = "Basketball";
+	_vm->_basketball->_court->_basketBall._objectType = kBall;
+	_vm->_basketball->_court->_basketBall._objectID = ballID;
+	_vm->_basketball->_court->_basketBall.center = ballLocation;
+	_vm->_basketball->_court->_basketBall._velocity = bellVelocity;
+	_vm->_basketball->_court->_basketBall.radius = radius;
+	_vm->_basketball->_court->_basketBall._collisionEfficiency = 1.0F;
+	_vm->_basketball->_court->_basketBall._friction = 0;
+	_vm->_basketball->_court->_basketBall._ignore = false;
+
+	_vm->_basketball->_court->_basketBall.save();
+
+	return 1;
+}
+
+int LogicHEBasketball::u32_userInitVirtualBall(U32FltPoint3D &ballLocation, U32FltVector3D &bellVelocity, int radius, int ballID) {
+	_vm->_basketball->_court->_virtualBall._description = "Virtual Basketball";
```

Comment:
Creating a reference to virtualBall could make the code a bit easier on the eyes, i.e.

```suggestion
        auto virtualBall = &_vm->_basketball->_court->_virtualBall;
	virtualBall._description = "Virtual Basketball";
```